### PR TITLE
Fixes bson macro for non sealed trait families

### LINKED ISSUE #29: Port ReactiveMongo/ReactiveMongo#862
https://github.com/ReactiveMongo/ReactiveMongo/pull/862

### PR DESCRIPTION
**Pull request by jtjeferreira**
fixes #29

### CODE REVIEW

**Comment by jtjeferreira** [comment]
Hi @cchantep. Could you help me with the error?
```
[error] /home/travis/build/ReactiveMongo/ReactiveMongo-BSON/api/src/test/scala/MacroSpec.scala:277:38: local val macroCfg$macro$345 in value $anonfun is never used
[error]       val format = Macros.handlerOpts[UT2, UnionType[UA2 \/ UB2] with AutomaticMaterialization]
```

**Comment by jtjeferreira** [comment]
Thanks for the fast response @cchantep! Tests are now green

File: api/src/test/scala/MacroSpec.scala [hunk 1]
```diff
@@ -283,6 +283,23 @@ final class MacroSpec extends org.specs2.mutable.Specification {
       }
     }
 
+    "handle union types without sealed trait" in {
```
Comment (cchantep) [inline api/src/test/scala/MacroSpec.scala]: Can be moved up (in the `>>` group) without repeating `handle union types` in description

Comment (jtjeferreira) [reply api/src/test/scala/MacroSpec.scala]: Done

**Pull request merged**